src/notebooks/controllers/preferredKernelConnectionService.node.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { NotebookDocument, workspace, Uri } from 'vscode';
+import * as path from '../../platform/vscode-path/resources';
+import { isParentPath } from '../../platform/common/platform/fileUtils';
+import { EnvironmentType } from '../../platform/pythonEnvironments/info';
+import { getEnvironmentType } from '../../platform/interpreter/helpers';
+import { Environment, ProposedExtensionAPI } from '../../platform/api/pythonApiTypes';
+
+export function findPreferredPythonEnvironment(
```

Comment:
Copy of the code from the file `preferredKernelConnectionService.ts`, but using the new Python extension API types,